vm-transfer/zvm-transfer.py

```diff
@@ -0,0 +1,494 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+"""
+A python script that clones a VM from the source system to the destination system
+"""
+
+__version__ = "0.1.0"
+
+import os
+import sys
+import logging
+import argparse
+import atexit
+import requests
+from pprint import pformat, pprint
+from munch import unmunchify
+
+sys.path.append('/opt/symphony-client')
+
+import symphony_client
+from config import Config
+
+LOGS_DIR = "/var/log"
+STRATO_LOGS_DIR = LOGS_DIR + "/stratoscale"
+ZVM_CLONE_LOGS_DIR = STRATO_LOGS_DIR + "/zvm-transfer"
+LOGGER_NAME = "zvm-transfer"
+VPSA_VOLUME_TEMPLATE = 'neokarm_volume-{}'
+
+DRY_RUN = Config.DEFAULT_IS_DRY_RUN
+
+logger = logging.getLogger(LOGGER_NAME)
+
+arguments = None
+
+
+def migrate_vm(vm_id_or_name):
+    logger.info("vm_id: %s", vm_id_or_name)
+
+    src_symp_client = init_src_symp_client()
+    vm_list = src_symp_client.vms.list(detailed=True)
+    vm = get_vm_by_id(src_symp_client, vm_id_or_name, vm_list)
+    if not vm:
+        vm = get_vm_by_name(src_symp_client, vm_id_or_name)
+        if not vm:
+            sys.exit(1)
+    vm_id = vm.id
+    if vm.status == 'active':
+        logger.info("VM {} ({}) is active in source cluster".format(vm.name, vm_id))
+        sys.exit(1)
+
+    dst_symp_client = init_dst_symp_client()
+    dest_vm = get_vm_by_name(dst_symp_client, vm.name)
+    if dest_vm:
+        logger.info("VM {} ({}) already exists in destination".format(dest_vm.name, dest_vm.id))
+        sys.exit(1)
+
+    networks = create_networking_map(src_symp_client, dst_symp_client, vm)
+    manageable_volumes, existing_volumes = check_volumes_in_dest(vm, dst_symp_client)
+
+    create_new_vm(vm, networks, manageable_volumes, existing_volumes, dst_symp_client)
+
+    logger.info("VM %s cloning is complete", vm.name)
+
+
+def get_vm_by_id(client, vm_id, vm_list=None):
+    if vm_list is None:
+        vm_list = client.vms.list(detailed=True)
+    filtered_vm_list = [vm for vm in vm_list if vm.id == vm_id]
+    if not filtered_vm_list:
+        logger.info("VM with id: %s does not exist", vm_id)
+        return None
+    return filtered_vm_list[0]
+
+
+def get_vm_by_name(client, vm_name, vm_list=None):
+    if vm_list is None:
+        vm_list = client.vms.list(detailed=True)
+    filtered_vm_list = [vm for vm in vm_list if vm.name == vm_name]
+    if not filtered_vm_list:
+        logger.info("VM with name: %s does not exists", vm_name)
+        return None
+    return filtered_vm_list[0]
+
+
+def init_src_symp_client():
+    my_session = requests.Session()
+    my_session.verify = False
+    client = symphony_client.Client(url='https://%s' % Config.SRC_CLUSTER_IP, session=my_session)
+    client.login(domain=Config.SRC_ACCOUNT,
+                 username=Config.SRC_USERNAME,
+                 password=Config.SRC_PASSWORD,
+                 project='default',
```

Comment:
I don't like the fact that 'default' project is hard-coded.

Besides that, it's confusing to me that SRC_PROJECT is not the project associated with SRC_ACCOUNT and SRC_USERNAME. I'm guessing that it will be confusing to anyone else using this script.

I suggest differentiating the constant names as follows:
SRC_ADMIN_ACCOUNT_NAME
SRC_ADMIN_USERNAME
SRC_ADMIN_PROJECT_NAME
SRC_TRANSFER_PROJECT_NAME

This comment also applies for the DST constants.